src/aiida/orm/nodes/data/code/portable.py

```diff
@@ -177,3 +178,31 @@ def filepath_executable(self, value: str) -> None:
             raise ValueError('The `filepath_executable` should not be absolute.')
 
         self.base.attributes.set(self._KEY_ATTRIBUTE_FILEPATH_EXECUTABLE, value)
+
+    @property
+    def filepath_files(self) -> pathlib.PurePosixPath:
+        """Return the absolute path of the directory that contains the code files.
+
+        :return: The directory of the code files required for the `filepath_executable`.
+        """
+        return pathlib.PurePosixPath(self.base.attributes.get(self._KEY_ATTRIBUTE_FILEPATH_FILES))
+
+    @filepath_files.setter
+    def filepath_files(self, value: t.Union[str, pathlib.Path]) -> None:
+        """Set the absolute path of the directory that contains the code files.
+
+        :param value: The absolute filepath of the directory of the uploaded code files.
+        """
+        type_check(value, (str, pathlib.Path))
+
+        if not pathlib.PurePosixPath(value).is_absolute():
+            raise ValueError('`filepath_files` must be an absolute path.')
+
+        if not pathlib.Path(value).exists():
+            raise ValueError('`filepath_files` must be an existing directory.')
+
+        self.base.attributes.set(self._KEY_ATTRIBUTE_FILEPATH_FILES, str(value))
```

Comment:
Why is this needed?

Reply:
Here, I just followed the same code that is implemented for `filepath_executable`. Good to have also `filepath_files` in the attributes when it is being set, just as for `filepath_executable`, no?

Reply:
Well for the filepath_executable there is a reason to store it in the attributes. You have to store the name of the executable somewhere. However, for `filepath_files`, the value passed is just the local path that contains the files. These are then copied to the repository. There is no point in keeping the original location of the directory from which the files were copied. When would you ever use that?

Reply:
I mean, it might still be useful information to store. But I agree it's not strictly necessary. I've removed it now, so this should be ready for final review. Cheers!